### PR TITLE
ADD: DIY hints content and hints promot behavior

### PR DESCRIPTION
**Pull request by hgfkeep**
add example : diy hints content and hints prompt behavior

this may helpful when some use rustyline do some cli real in specific domains.

### CODE REVIEW

File: examples/diy_hints.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+use rustyline::{highlight::Highlighter, hint::Hinter, Context};
+use rustyline::Editor;
+use rustyline_derive::{Completer, Helper, Validator};
+
+#[derive(Completer, Helper, Validator)]
+struct DIYHinter {
+    hints: HashSet<String>,
+}
+
+impl Highlighter for DIYHinter {
```
Comment (gwenn) [inline examples/diy_hints.rs]: You can remove this `impl` block and derive `Highlighter` instead.

File: examples/diy_hints.rs [hunk 2]
```diff
@@ -0,0 +1,61 @@
+use std::collections::HashSet;
+
+use rustyline::{highlight::Highlighter, hint::Hinter, Context};
+use rustyline::Editor;
+use rustyline_derive::{Completer, Helper, Validator};
+
+#[derive(Completer, Helper, Validator)]
+struct DIYHinter {
+    hints: HashSet<String>,
```
Comment (gwenn) [inline examples/diy_hints.rs]: A radix trie may be better.

File: examples/diy_hints.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    println!("This is a DIY hint hack of rustyline");
+    let h = DIYHinter {
+        hints: HashSet::new(),
+    };
+    let mut rl = Editor::new();
+    rl.set_helper(Some(h));
+    rl.helper_mut().expect("No helper").hints = diy_hints();
+
+    loop {
+        let passwd = rl.readline("> ")?;
```
Comment (gwenn) [inline examples/diy_hints.rs]: `passwd` should be renamed.

**Comment by hgfkeep** [comment]
thanks @gwenn review, I'll update it.

**Pull request closed without merging**